File: go_solution/001_010/twoSum/twoSum.go
package twoSum
// 1. Two Sum
func twoSum(nums []int, target int) []int {
	trace := make(map[int]int)
	for idx, v := range nums {
		if k, ok := trace[target-v]; ok {
			return []int{k, idx}
		}
		if _, ok := trace[v]; !ok {
			trace[v] = idx
		}
	}
	return []int{}
}

File: python_solution/001_010/TwoSum.py
# 001. Two Sum

class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        res = []
        if not nums:
            return res
        dict = {}
        for i in range(len(nums)):
            complement = target - nums[i]
            if dict.get(complement):
                return [dict[complement], i]
            dict[nums[i]] = i

        return res